src/rust/uk.mangainua/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+
+	let mut is_nsfw = false;
+	let mut status = String::from("Unknown");
+
+	let mut categories = Vec::new();
+	for categ in html.select(".item__full-sidebar--description a").array(){
+		let categ_node = categ.as_node();
+		let name = categ_node.text().read();
+
+		if categories.contains(&name){ // categories are duplicated on the website
+			continue;
+		}
+		categories.push(name.clone());
+
+		if !is_nsfw { // check if only didnt find nsfw category
+			is_nsfw = helper::is_nsfwbool(name.clone());
+		}
+		if status == "Unknown" { // check if only didnt find status
+			status = helper::get_status_string(name.clone());
+		}
+	}
+
+	let mut status_res = MangaStatus::Unknown;
+	if status == "Ongoing"{
+		status_res = MangaStatus::Ongoing;
+	}
+	else if status == "Unknown"{
+		status_res = MangaStatus::Unknown;
+	}
+	else if status == "Completed"{
+		status_res = MangaStatus::Completed;
+	}
+
+	let manga = Manga {
+		id,
+		cover,
+		title,
+		author : String::new(),
+		artist : String::new(),
+		description,
+		url : String::new(),
+		categories: categories.clone(),
+		status: status_res,
+		nsfw: if is_nsfw {MangaContentRating::Nsfw} else {MangaContentRating::Safe},
+		viewer: MangaViewer::Rtl,
+	};
+	Ok(manga)
+}
+
+#[get_chapter_list]
+fn get_chapter_list(id: String) -> Result<Vec<Chapter>> {
+	let html = Request::new(id.as_str(), HttpMethod::Get).html();
+	
+	let mut chapter : f32 = -1.0;
+	
+	let mut res: Vec<Chapter> = Vec::new();
+	for result in html.select(".linkstocomicsblockhidden .ltcitems").array() {
+		let res_node = result.as_node();
+		let href = res_node.select("a").attr("href").read();
+		let mut title = res_node.select("a").text().read();
+
+		let date = res_node.select(".ltcright").first().text()
+				.0
+				.as_date("dd.MM.yyyy", None, None)
+				.unwrap_or(-1.0);
+
+		let scanlator = res_node.select(".ltcright .tooltip .tooltiptext").text().read();
+
+		// Alternative translation don't have chapter number. use previous parsed
+		if title.contains("Альтернативний переклад"){
+			title = String::from("↑") + &title;
+		}
+		else
+		{	
+			// parse volume and chapter.
+			let volume_chapter = title.clone();
+			let replaced = volume_chapter.replace("НОВЕ ", "");
+			let arr: Vec<_> = replaced.split_whitespace().collect();
+
+			match arr[3].parse::<f32>(){
+				Ok(n) => chapter = n,
+				_e => continue,
+			};
+		}
+
+		let chapter = Chapter {
+			id: href.clone(),
+			title,
+			volume: -1.0,
+			chapter,
+			url: href.clone(),
+			date_updated: date,
+			scanlator,
+			lang: String::from("uk"),
+		};
+		res.push(chapter);
+	}
+	
+	Ok(res)
+}
+
+#[get_page_list]
+fn get_page_list(id: String) -> Result<Vec<Page>> {
+	let html = Request::new(id.as_str(), HttpMethod::Get).html();
+
+	let mut pages: Vec<Page> = Vec::new();
+	//for result in html.select(".loadcomicsimages img").array() {
+	for (index, result) in (0_i32..).zip(html.select(".loadcomicsimages img").array()) {
```

Comment:
not sure what you're doing with the zip thing here, just do `.enumerated()` (it does the same thing)

Reply:
Clippy told me to do so